samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -131,7 +133,20 @@ def __enter__(self):
                                                       self._debug_args,
                                                       self._debugger_path)
 
-        self._check_docker_connectivity()
+        container_manager = self._get_container_manager(self._docker_network, self._skip_pull_image)
+
+        if not container_manager.is_docker_reachable:
+            raise InvokeContextException("Running AWS SAM projects locally requires Docker. Have you got it installed?")
+
+        if self._container_name:
+            if not container_manager.is_valid_container_name(self._container_name):
+                raise InvokeContextException("'{}' is not a valid Docker container name"
+                                             "only [a-zA-Z0-9][a-zA-Z0-9_.-] are allowed.".format(self._container_name))
+
+            if container_manager.is_container_name_taken(self._container_name):
```

Comment:
This is going to add extra overhead to the invoke call. Instead of checking here, can we let the Docker create API throw an exception, catch it there, and surface the error to customers? 

I would expect the common case to be the one where container name is available. So this will optimize for the common case.

Reply:
There is no problem eliminating all of the check s  upfront, and that will boost performance  in common case, agreed. I'll  remove it

My thinking for that was: "If you can avoid exceptions, do that" :)

Reply:
Makes sense.

Reply:
Eliminated all checks